lib/ruby_indexer/lib/ruby_indexer/augmentation.rb

```diff
@@ -0,0 +1,24 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyIndexer
+  module Augmentation
```

Comment:
Why do we use mixin here instead of inheritance, like `Addon`?

Reply:
Because we're not providing any functionality to enhancements at the moment. We just need them to comply to a specific interface.